### ISSUE TITLE
How to handle a list of string

### ISSUE THREAD

**Issue by vincent-paing** [SiimKinks/sqlitemagic#8]
Currently, I have the following field in my entity

```java
@Table(persistAll = true, value = "table_movie") @AutoValue public abstract class MovieEntity {

  @Id @Column(value = "movie_id") public abstract long id();

  @Column(value = "movie_name") public abstract String name();

  @Column(value = "category") public abstract CategoryEntity category();

  @Column(value = "alt_names") public abstract List<String> alternativeNames();
}
```

I have added a Transformer class for List String that convert into JSON vice versa.

```java
@Transformer public class StringListTransformer {

  @ObjectToDbValue public static String objectToDBValue(List<String> object) {
    return new Gson().toJson(object);
  }

  @DbValueToObject public static List<String> dbValueToObject(String dbValue) {
    return new Gson().fromJson(dbValue, new TypeToken<List<String>>() {
    }.getType());
  }
}
```

When I compile my project it gave out this error with no instructions to recover

> Error:Execution failed for task ':sqlitemagic:compileDebugJavaWithJavac'.
> java.lang.ClassCastException: com.sun.tools.javac.code.Type$ClassType cannot be cast to javax.lang.model.type.ArrayType

EDIT: Seems like it's because of The Transformer class, when I remove it and the list field, it compile successfully.

**Comment by SiimKinks** [comment]
This is a bug in transformer method types parsing.
You can replace List<String> with String[] until this issue is fixed.

**Comment by vincent-paing** [comment]
AutoValue only support primitive arrays so using String[] with AutoValue will not work. For now, I will just use without AutoValue

**Comment by SiimKinks** [comment]
This bug is fixed in version 0.17.0
